src/libraries/Common/src/System/Security/Cryptography/MLKem.cs

```diff
@@ -137,6 +128,84 @@ public void Encapsulate(Span<byte> ciphertext, Span<byte> sharedSecret)
             EncapsulateCore(ciphertext, sharedSecret);
         }
 
+        /// <summary>
+        ///   Creates an encapsulation ciphertext and shared secret, writing them into the provided buffers.
+        /// </summary>
+        /// <param name="ciphertext">
+        ///   The buffer to receive the ciphertext.
+        /// </param>
+        /// <param name="sharedSecret">
+        ///   The buffer to receive the shared secret.
+        /// </param>
+        /// <param name="ciphertextWritten">
+        ///   When this method returns, the total number of bytes written into <paramref name="ciphertext"/>.
+        /// </param>
+        /// <param name="sharedSecretWritten">
+        ///   When this method returns, the total number of bytes written into <paramref name="sharedSecret"/>.
+        /// </param>
+        /// <exception cref="CryptographicException">
+        ///   <para>An error occurred during encapsulation.</para>
+        ///   <para>-or -</para>
+        ///   <para><paramref name="ciphertext"/> overlaps with <paramref name="sharedSecret"/>.</para>
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///   <para><paramref name="ciphertext" /> is too small to hold the ciphertext.</para>
+        ///   <para> -or- </para>
+        ///   <para><paramref name="sharedSecret" /> is too small to hold the shared secret.</para>
+        /// </exception>
+        /// <exception cref="ObjectDisposedException">The object has already been disposed.</exception>
+        public void Encapsulate(
+            Span<byte> ciphertext,
+            Span<byte> sharedSecret,
+            out int ciphertextWritten,
+            out int sharedSecretWritten)
```

Comment:
API Review will probably want `Bytes` inserted in there somewhere.  e.g. ciphertextBytesWritten; just because we have the `out {bytes,chars,items}Written` guidance.

If you want to leave it out for now and argue about it there, that's fine :smile: